chapters/Index.md

```diff
@@ -1,58 +1,3095 @@
+Document:: according to the type of the identifier. If the identifier
+encodes a 33-byte compressed secp256k1 public key, the ::Initial DID
+Document:: can be generated by following [Algo 5. Deterministically
+Generate Initial DID Document]. Otherwise, the identifier MUST encode a
+32-byte hash of a ::Genesis Document::, in this case first retrieve the
+::Genesis Document:: using [Algo 6. Retrieve Genesis Document] and then
+verify this document according to [Algo 7. Verify Genesis Document].
+
+Second, iterate through ::Beacon Signals:: posted to the Bitcoin blockchain
+using [Algo 8. Process Beacon Signals] processing them according to their
+::Beacon Type::. Processing ::Beacon Signals:: retrieves the announced
+::BTCR2 Update:: for the DID being resolved. The algorithms for processing
+::Beacon Signals:: according to their ::Beacon Types:: are as follows: for
+::Singleton Beacons:: use [Algo 9. Process Singleton Beacon Signal];  for
+::Map Beacons:: use [Algo 10. Process Map Beacon Signal]; for ::SMT
+Beacons:: use [Algo 11. Process SMT Beacon Signal].
+
+::BTCR2 Updates:: announced with ::Beacon Signals:: are then applied to the
+DID document. ::BTCR2 Updates:: MUST be invocations of the capability to
+update the DID being resolved. To verify this, the resolver MUST first
+derive the root capability from the DID itself using  [Algo 12. Derive Root
+Capability], then verify and apply the invocation using [Algo 13. Apply
+BTCR2 Update].
+
+Traverse the blockchain and apply updates in temporal order, updating and
+maintaining a ::Contemporary DID Document:: until all updates are
+processed. At each stage, the Contemporary DID Document defines the ::BTCR2
+Beacons:: active at that point in time; those Beacons – and only those
+Beacons – are checked for updates.
+
+Continue traversing the chain until all timely, authentic ::Beacon
+Signals:: have been processed and their announced updates have been applied
+in order.
+
+The resulting DID document is the canonical DID Document and MUST be
+returned to the caller of the resolver function in a resolution response
+which can be constructed following [Algo 14. Construct Resolution Result].
+
+NOTE. If there are no updates, the ::Initial DID Document:: is returned.
+
+#### Algo 4. Process Resolution Inputs {.tabbed .unnumbered}
+
+This algorithm processes the resolution inputs from a resolution request.
+First, parse the identifier string, then process the resolution options.
+
+First take the identifier string, validate it according to the [DID
+syntax](https://www.w3.org/TR/did-1.1/#did-syntax) and decode it to
+retrieve the identifier type, version, network and ::Genesis Bytes::.
+
+The identifier string MUST consist of three components separated by a ‘:’
+character. The first component MUST be the string "did", the second
+component MUST be "btcr2" and the third component is the BTCR2-specific
+identifier. Otherwise an INVALID_DID error MUST be raised.
```

Comment:
```suggestion
The identifier string MUST consist of three components separated by a ‘:’
character. The first component MUST be the string "did", the second
component MUST be "btcr2" and the third component is the BTCR2-specific
identifier. If the second component is not "btcr2" a METHOD_NOT_SUPPORTED error 
MUST be raised, otherwise for all other errors an INVALID_DID error MUST be raised.
```

Missing the METHOD_NOT_SUPPORTED case